Sources/SymDiff/source/Transform.cs

```diff
@@ -481,14 +481,29 @@ private static string sanitize(string s)
     { 
       return s.Replace("$", "");
     }
+
+    private static (string, string) RemoveCommonPrefix(string s1, string s2)
+    {
+      int i = 0;
+      while (i < Math.Min(s1.Length, s2.Length) && s1[i] == s2[i]) {
+        ++i;
+      }
+      return (s1.Substring(i), s2.Substring(i));
+
+    }
+
     public static string mkEqProcName(string p1, string p2)
     {
-      var p1WithoutPrefix = p1.Split(".").Last();
-      var p2WithoutPrefix = p2.Split(".").Last();
-      if (p1WithoutPrefix.Equals(p2WithoutPrefix))
+      var p1Parts = p1.Split(".");
+      var p2Parts = p2.Split(".");
+      var p1WithoutPrefix = p1Parts.Last();
+      var p2WithoutPrefix = p2Parts.Last();
+      var (p1Different, p2Different) = RemoveCommonPrefix(p1WithoutPrefix, p2WithoutPrefix);
+      if (p1Different.Length == 0 && p2Different.Length == 0) {
         return "EQ_" + sanitize(p1WithoutPrefix);
-      else
-        return "EQ_" + sanitize(p1) + "__xx__" + sanitize(p2);
+      } else {
+        return "EQ_" + sanitize(p1Different) + "__xx__" + sanitize(p2Different);
```

Comment:
There is a risk here because we don't ever want the eq names to clash. So, it seems that this would emit the same eq name for `(p1 = XXX.AddTwo, p2 = XXX.AddTwoDirect)` and `(p1' = XXX.SubTwo, p2' = XXX.SubTwoDirect)`. If so, this is a bug.

Reply:
Specifically both cases emit the final string `EQ_Direct`, assuming sanitize is a no-op.

Reply:
Yep, you're right. I had a vague memory that I'd thought of doing it this way and decided not to, and now I remember that it was for exactly that reason. :)